apps/studio/components/interfaces/RoleImpersonationSelector/RoleImpersonationSelector.tsx

```diff
@@ -0,0 +1,133 @@
+import { useState } from 'react'
+
+import { PostgrestRole } from 'lib/role-impersonation'
+import { useRoleImpersonationStateSnapshot } from 'state/role-impersonation-state'
+import { DropdownMenuSeparator, cn } from 'ui'
+import { AnonIcon, AuthenticatedIcon, ServiceRoleIcon } from './Icons'
+import RoleImpersonationRadio from './RoleImpersonationRadio'
+import UserImpersonationSelector from './UserImpersonationSelector'
+
+export interface RoleImpersonationSelectorProps {
+  serviceRoleLabel?: string
+  padded?: boolean
+}
+
+const RoleImpersonationSelector = ({
+  serviceRoleLabel,
+  padded = true,
+}: RoleImpersonationSelectorProps) => {
+  const state = useRoleImpersonationStateSnapshot()
+
+  const [selectedOption, setSelectedOption] = useState<PostgrestRole | undefined>(() => {
+    if (
+      state.role?.type === 'postgrest' &&
+      (state.role.role === 'anon' || state.role.role === 'authenticated')
+    ) {
+      return state.role.role
+    }
+
+    return 'service_role'
+  })
+
+  const isAuthenticatedOptionFullySelected = Boolean(
+    selectedOption === 'authenticated' &&
+      state.role?.type === 'postgrest' &&
+      state.role.role === 'authenticated' &&
+      state.role.user
+  )
+
+  function onSelectedChange(value: PostgrestRole) {
+    if (value === 'service_role') {
+      // do not set a role for service role
+      // as the default role is the "service role"
+      state.setRole(undefined)
+    }
+
+    if (value === 'anon') {
+      state.setRole({
+        type: 'postgrest',
+        role: value,
+      })
+    }
+
+    setSelectedOption(value)
+  }
+
+  return (
```

Comment:
this is probably more of a feedback - am just wondering from a fresh user's POV, i'm on the table editor - i'm not too sure what choosing the role does for me. "Database connection settings" feels like the wrong terminology here too. Might need some more description in this popover?

![image](https://github.com/supabase/supabase/assets/19742402/a3b47d1d-61d3-4c27-b055-60a94e8ff3d6)

Reply:
"Database role settings"

Reply:
Updated 👍🏻